src/hooks/useConnectOwnedVenues.tsx

```diff
@@ -0,0 +1,88 @@
+import { useMemo, useCallback } from "react";
+import { ReactHook } from "types/utility";
+import { AnyVenue } from "types/venues";
+
+import { WithId, withId } from "utils/id";
+import { ownedVenuesDataSelector } from "utils/selectors";
+
+import { useFirestoreConnect, isLoaded } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+
+export const useConnectOwnedVenues = (userId?: string) => {
+  useFirestoreConnect(
+    userId
+      ? {
+          collection: "venues",
+          where: [["owners", "array-contains", userId]],
+          storeAs: "ownedVenues",
+        }
+      : undefined
+  );
+};
+
+export interface UseOwnedVenuesProps {
+  currentVenueId?: string;
+}
+
+export interface UseOwnedVenuesData {
+  isLoaded: boolean;
+  isLoading: boolean;
+
+  currentVenue?: WithId<AnyVenue>;
+
+  ownedVenues: WithId<AnyVenue>[];
+  ownedVenuesIds: string[];
+
+  findVenueInOwnedVenues: (
+    searchedForVenueId: string
+  ) => WithId<AnyVenue> | undefined;
+}
+
+export const useOwnedVenues: ReactHook<
+  UseOwnedVenuesProps,
+  UseOwnedVenuesData
+> = ({ currentVenueId }): UseOwnedVenuesData => {
+  const { userId } = useUser();
+  useConnectOwnedVenues(userId);
+
+  const venues = useSelector(ownedVenuesDataSelector);
+
+  const ownedVenuesIds = useMemo(() => Object.keys(venues ?? {}), [venues]);
+
+  const ownedVenues = useMemo(
+    () => Object.entries(venues ?? {}).map(([id, venue]) => withId(venue, id)),
+    [venues]
+  );
+
+  const findVenueInOwnedVenues = useCallback(
+    (searchedForVenueId: string) => {
+      const found = venues?.[searchedForVenueId];
+      return found ? withId(found, searchedForVenueId) : undefined;
+    },
+    [venues]
+  );
```

Comment:
```suggestion
  const ownedVenuesData = useSelector(ownedVenuesDataSelector);
  const ownedVenues = useSelector(ownedVenuesSelector);
  
  const ownedVenuesIds = useMemo(() => Object.keys(venues ?? {}), [venues]);

  const getVenueInOwnedVenues = useCallback(
    (searchedForVenueId: string) => {
      const foundVenue = venues?.[searchedForVenueId];
      return foundVenue ? withId(found, searchedForVenueId) : undefined;
    },
    [venues]
  );
```

Reply:
` useSelector(ownedVenuesSelector);` doesn't return `WithId<>` objects I think it's better not to proliferate non-id values

Reply:
@goran-peoski-work it should return `WithId`

![image](https://user-images.githubusercontent.com/63194656/126542764-06ce582c-9ff3-4534-9f4d-dfd4c7ad547e.png)